modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ConfigurationExpirationMonitor.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.manager;
+
+import org.mule.api.lifecycle.Startable;
+import org.mule.api.lifecycle.Stoppable;
+
+/**
+ * Monitors configuration instances looking for those which should be expired.
+ *
+ * It only exposes {@link #start()} and {@link #stop()} method to signal
+ * when monitoring should start/stop.
+ *
+ * @since 4.0
+ */
+public interface ConfigurationExpirationMonitor extends Startable, Stoppable
```

Comment:
how would you do it? You need a component with that responsibility and you need to be able to start and stop it